scalableminds/webknossos
# Preserve certain state properties on some sites during reload and navigation

This PR aims to preserve certain state properties, like a search query or a drop down selection, during a page reload and back/forward browser navigation.
I've implemented this for a couple of our admin sites and wanted to request feedback first, before adding it to all of the lists :)

This PR also adds libdefs for react-router and libdef stubs for Antd. Before this we always lost prop type information of components when using the `withRouter` or `Form.create` wrappers.
I've also removed the libdefs for backbone and marionette which brought a fascinating speed-up for flows initial type checking (down from >1min to ~10s :tada:), although we didn't even use them.

If you agree with the method I used to preserve those state properties, I'll continue to add this to the other admin pages.

The issue also demands a refresh button for our lists. I'm not sure whether that's worth it, considering that a browser reload is pretty fast. What do you think?

### Mailable description of changes:
 - Search criteria of all lists are now preserved during page reloads and forward/backward browser navigation.

### Steps to test:
- Open the task list and select a user and press search or/and enter a search term. Reloading the page should keep the entered selections/search terms. Navigating somewhere else and using the back button should as well.
- Same for the search queries in the dataset and projects lists.
- In the explorative tracings list, it should additionally be remembered whether archived or open annotations are being displayed.

### Issues:
- fixes #2265

------
